external-game-data.js

```diff
@@ -0,0 +1,101 @@
+/*
+  ==================================
+  EXTERNAL GAME DATA MOD (@mildmojo)
+  ==================================
+
+  Load your Bitsy game data from an external file or URL, separating it from your
+  (modified) Bitsy HTML.
+
+  Usage: IMPORT <file or URL>
+
+  Examples: IMPORT frontier.bitsydata
+            IMPORT http://my-cool-website.nz/frontier/frontier.bitsydata
+            IMPORT /games/frontier/data/frontier.bitsydata
+
+  HOW TO USE:
+    1. Copy-paste this script into a new script tag after the Bitsy source code.
+    2. Copy all your Bitsy game data out of the script tag at the top of your
+       HTML into another file (I recommend `game-name.bitsydata`). In the HTML
+       file, replace all game data with a single IMPORT statement that refers to
+       your new data file.
+
+  NOTE: Chrome can only fetch external files when they're served from a
+        web server, so your game won't work if you just open your HTML file from
+        disk. You could use Firefox, install a web server, or, if you have
+        development tools like NodeJS, Ruby, Python, Perl, PHP, or others
+        installed, here's a big list of how to use them to serve a folder as a
+        local web server:
+        https://gist.github.com/willurd/5720255
+
+        If this mod finds an IMPORT statement anywhere in the Bitsy data
+        contained in the HTML file, it will replace all game data with the
+        IMPORTed data. It will not execute nested IMPORT statements in
+        external files.
+
+  Version: 1.0
+  Bitsy Version: 4.5, 4.6
+  License: WTFPL (do WTF you want)
+*/
+
+// Give a hoot, don't pollute; encapsulate in an IIFE for isolation.
+(function(globals) {
+  'use strict';
+
+  var _load_game = load_game;
+  globals.load_game = function(game_data, startWithTitle) {
+    tryImportGameData(game_data, function withGameData(err, importedData) {
+      if (err) {
+        console.warn('Make sure game data IMPORT statement refers to a valid file or URL.');
+        throw err;
+      } else {
+        _load_game(dos2unix(importedData), startWithTitle);
+      }
+    });
+  };
+
+  function tryImportGameData(gameData, done) {
+    // Make sure this game data even uses an IMPORT statement.
+    if (gameData.indexOf('IMPORT') === -1) {
+      return done('No IMPORT found in Bitsy data. See instructions for external game data mod.', gameData);
+    }
+
+    var trim = function(line) { return line.trim(); };
+    var isImport = function(line) { return getType(line) === 'IMPORT'; };
+    var importCmd = gameData
+      .split("\n")
+      .map(trim)
+      .find(isImport);
+    var src = (importCmd || '').split(/\s+/)[1];
+
+    if (src) {
+      return fetchData(src, done);
+    } else {
+      return done('IMPORT missing a URL or path to a Bitsy data file!');
+    }
+  }
```

Comment:
I refactored this function so the code's less noisy; no more for loop searching the file, and a guard statement to make sure it's even worth looking for IMPORT.